# Σύγκριση Σχεδίων Αποπληρωμής Φοιτητικού Δανείου με την PROC LOAN

## Περίληψη Διοίκησης

Ένα γραφείο οικονομικής βοήθειας αξιολογεί πώς μια αποφοιτούσα ομάδα φοιτητών θα πρέπει να αποπληρώσει ένα αντιπροσωπευτικό υπόλοιπο $27,500 συγκρίνοντας τέσσερις δομές αποπληρωμής — ένα ομοσπονδιακό σταθερού επιτοκίου βασικό σχέδιο, μια ιδιωτική αναχρηματοδότηση με προμήθεια έναρξης, ένα κυμαινόμενου επιτοκίου (ARM) δάνειο με ανώτατο όριο, και μια επιδοτούμενη από τον εργοδότη μείωση επιτοκίου (buydown) — χρησιμοποιώντας την `PROC LOAN`. Σε διάρκεια 120 μηνών, οι τέσσερις προσφορές ευθυγραμμίζονται καθαρά ως προς τη μηνιαία δόση και τον συνολικό τόκο στα αρχικά τους επιτόκια: το ομοσπονδιακό βασικό σχέδιο κοστίζει τον περισσότερο τόκο (**$10,021.22** στο 6.53%, δόση **$312.68**), η ιδιωτική αναχρηματοδότηση μειώνει το επιτόκιο αλλά προσθέτει προμήθεια $412.50 (**$9,120.20** στο 5.99%, δόση **$305.17**), και το χαμηλότερα εκτιμώμενο ARM (**$7,099.76** στο 4.75%, δόση **$288.33**) και η μείωση επιτοκίου από τον εργοδότη (**$6,700.67** στο 4.50%, δόση **$285.01**) έχουν τους μικρότερους λογαριασμούς τόκων. Ένα μπλοκ `COMPARE` στη συνέχεια αναφέρει τον σωρευτικό τόκο, το κεφάλαιο και το ανεξόφλητο υπόλοιπο κάθε σχεδίου στους 36, 60 και 120 μήνες, δείχνοντας πόσο έχει αποσβεστεί κάθε δάνειο στα σημεία όπου ένας δανειολήπτης είναι πιο πιθανό να αναχρηματοδοτήσει ή να εξοφλήσει.

## Πηγές Δεδομένων

| Σύνολο Δεδομένων | Γραμμές | Περιγραφή | Βασικές Μεταβλητές |
|---------|------|-------------|---------------|
| `borrowers` | 40 | Συνθετικά προφίλ δανείων μιας αποφοιτούσας ομάδας, δημιουργημένα εσωτερικά με `call streaminit(20260531)` και `rand('uniform')`. Χρησιμοποιούνται για να τεκμηριώσουν ρεαλιστικούς όρους δανείου για τη σύγκριση. | `student_id` (1001-1040), `balance` (κεφάλαιο κατά την αποφοίτηση· αυτό το δείγμα κυμαίνεται από $11,800 έως $47,750, μέσος όρος $30,771), `apr` (ονομαστικό ετήσιο επιτόκιο 4.10%-9.10%, μέσος όρος 6.50%), `term` (120 ή 180 μήνες, μέσος όρος 144), `origination_pct` (προμήθεια 1.0%-2.0%, μέσος όρος 1.50%) |

Ο αντιπροσωπευτικός δανειολήπτης που αναλύεται με την `PROC LOAN` (ποσό $27,500, διάρκεια 120 μηνών, έναρξη Ιούλιος 2026) βρίσκεται κοντά στο κάτω-μεσαίο τμήμα της κατανομής υπολοίπων και επιτοκίων αυτής της ομάδας· δεν χρησιμοποιούνται εξωτερικά δεδομένα ή δεδομένα δικτύου. Η ομάδα υπάρχει για να τεκμηριώσει εύλογους όρους δανείου — η απευθείας σύγκριση εκτελείται στο ενιαίο αντιπροσωπευτικό δάνειο.

# Σύγκριση Σχεδίων Αποπληρωμής Φοιτητικού Δανείου με την PROC LOAN

Όταν οι φοιτητές αποφοιτούν, ένα γραφείο οικονομικής βοήθειας πρέπει να τους βοηθήσει να επιλέξουν ανάμεσα σε ανταγωνιστικές προσφορές αποπληρωμής. Η `PROC LOAN` (SAS/ETS) έχει σχεδιαστεί ακριβώς για αυτό: αποσβένει δάνεια σταθερού επιτοκίου, κυμαινόμενου επιτοκίου (ARM) και μείωσης επιτοκίου (buydown), και στη συνέχεια τα συγκρίνει ένα προς ένα ως προς τη δόση, τον συνολικό τόκο και την πρόοδο απόσβεσης.

Σε αυτό το notebook:

1. Δημιουργούμε μια συνθετική αποφοιτούσα ομάδα για να τεκμηριώσουμε ρεαλιστικούς όρους δανείου.
2. Συνοψίζουμε την ομάδα με την `PROC MEANS`.
3. Χτίζουμε τέσσερα σχέδια αποπληρωμής για ένα αντιπροσωπευτικό υπόλοιπο $27,500 και τα συγκρίνουμε με την `PROC LOAN` (FIXED / ARM / BUYDOWN + COMPARE).
4. Επανεκτελούμε το συνιστώμενο σταθερό σχέδιο μεμονωμένα για να επιβεβαιώσουμε τα αυτοτελή οικονομικά του στοιχεία.

Όλα εκτελούνται τοπικά σε ενσωματωμένα συνθετικά δεδομένα — χωρίς εξωτερικά αρχεία ή πρόσβαση σε δίκτυο.

## Βήμα 1 — Δημιουργία συνθετικής αποφοιτούσας ομάδας

Προσομοιώνουμε 40 δανειολήπτες. Καθένας τραβά ένα αρχικό κεφάλαιο κατά την αποφοίτηση, ένα APR χαλαρά συνδεδεμένο με την πιστωτική ποιότητα, μια τυπική διάρκεια αποπληρωμής (10 ή 15 χρόνια), και ένα ποσοστό προμήθειας έναρξης. Ο σπόρος (seed) κάνει τα δεδομένα αναπαραγώγιμα.

In [1]:
ΔΕΔΟΜΕΝΑ borrowers;
   CALL streaminit(20260531);
   LENGTH plan $ 28;
   ΕΠΑΝΑΛΗΨΗ student_id = 1001 ΕΩΣ 1040;
      /* Κεφάλαιο κατά την αποφοίτηση: $9,500 - $48,000 */
      balance = round(9500 + 38500*rand('uniform'), 50);
      /* Ετήσιο ονομαστικό APR ανά πιστωτική βαθμίδα: 4.0% - 9.5% */
      apr = round(4.0 + 5.5*rand('uniform'), 0.05);
      /* Τυπική διάρκεια αποπληρωμής: 120 ή 180 μήνες */
      ΕΑΝ rand('uniform') < 0.6 ΤΟΤΕ term = 120;
      ΑΛΛΙΩΣ term = 180;
      /* Προμήθεια έναρξης ως ποσοστό του κεφαλαίου: 1.0% - 2.0% */
      origination_pct = round(1.0 + rand('uniform'), 0.10);
      ΕΞΟΔΟΣ;
   ΤΕΛΟΣ;
ΕΚΤΕΛΕΣΗ;


NOTE: DATA borrowers


NOTE: Wrote borrowers (40 rows, 6 columns).
NOTE: DATA elapsed:
  wall  0.00 seconds
  cpu   0.00 seconds


## Βήμα 2 — Προφίλ της ομάδας

Πριν μοντελοποιήσουμε μεμονωμένα δάνεια, εξετάζουμε την κατανομή των υπολοίπων, των επιτοκίων και των διαρκειών. Αυτό μας δείχνει πώς μοιάζει ένας *αντιπροσωπευτικός* δανειολήπτης — η βάση για τη σύγκριση ένα προς ένα που ακολουθεί.

In [2]:
ΔΙΑΔΙΚΑΣΙΑ ΜΕΣΟΙ ΔΕΔΟΜΕΝΑ=borrowers n mean MIN MAX maxdec=2;
   ΜΕΤΑΒΛΗΤΗ balance apr term origination_pct;
   ΕΤΙΚΕΤΑ balance='Υπόλοιπο Δανείου ($)'
         apr='Ετήσιο Επιτόκιο APR (%)'
         term='Διάρκεια (μήνες)'
         origination_pct='Προμήθεια Έναρξης (%)';
ΕΚΤΕΛΕΣΗ;

                                                  The MEANS Procedure

 Variable         Label                                        N           Mean     Minimum     Maximum
 ------------------------------------------------------------------------------------------------------
 balance          Υπόλοιπο Δανείου ($)                        40       30771.25    11800.00    47750.00
 apr              Ετήσιο Επιτόκιο APR (%)                     40           6.50        4.10        9.10
 term             Διάρκεια (μήνες)                            40         144.00      120.00      180.00
 origination_pct  Προμήθεια Έναρξης (%)                       40           1.50        1.00        2.00
 ------------------------------------------------------------------------------------------------------




NOTE: PROC MEANS
NOTE: PROC MEANS statement used.


## Βήμα 3 — Σύγκριση τεσσάρων σχεδίων αποπληρωμής για έναν αντιπροσωπευτικό δανειολήπτη

Χρησιμοποιώντας ένα αντιπροσωπευτικό υπόλοιπο $27,500 σε διάρκεια 120 μηνών (10 ετών) με έναρξη τον Ιούλιο του 2026, παραθέτουμε τέσσερις ρεαλιστικές προσφορές:

- **Ομοσπονδιακό Άμεσο Χωρίς Επιδότηση (Βασικό)** — ένα απλό δάνειο σταθερού επιτοκίου στο 6.53%.
- **Ιδιωτική Αναχρηματοδότηση (με προμήθεια)** — ένα χαμηλότερο σταθερό επιτόκιο 5.99%, αλλά με κόστος έναρξης $412.50 (`INIT=`).
- **Ιδιωτικό Κυμαινόμενο ARM** — ένα κυμαινόμενο δάνειο 4.75% με όριο 1% ανά αναπροσαρμογή / 5% εφ' όρου ζωής (`CAPS=`), ανώτατο επιτόκιο 9.75% (`MAXRATE=`), ετήσια αναπροσαρμογή (`ADJUSTFREQ=`), και τη λέξη-κλειδί `WORSTCASE`.
- **Μείωση Επιτοκίου Εργοδότη 2-1** — μια επιδοτούμενη έναρξη στο 4.50% που, σύμφωνα με το αναφερόμενο πρόγραμμα, αυξάνεται μέσω `BDRATES=` στο 5.50% τον 2ο χρόνο και στο 6.50% στη συνέχεια.

Η δήλωση `COMPARE` ζητά τη συγκριτική προβολή μεταξύ δανείων στους 36, 60 και 120 μήνες, με φορολογικό συντελεστή 22% (`TAXRATE=`) και ελάχιστο ελκυστικό ποσοστό απόδοσης 4% (`MARR=`)· τα `OUTSUM=` και `OUTCOMP=` καταγράφουν την περίληψη ανά δάνειο και τις γραμμές σύγκρισης. Η παρακάτω λίστα αναφέρει, για κάθε σχέδιο και κάθε σημείο ελέγχου, τον **σωρευτικό τόκο που καταβλήθηκε, το σωρευτικό κεφάλαιο που εξοφλήθηκε, και το ανεξόφλητο υπόλοιπο** — μια σαφή εικόνα προόδου απόσβεσης μεταξύ των υποψήφιων σχεδίων.

> **Σημείωση για τις αναπροσαρμογές επιτοκίου.** Η `PROC LOAN` του Jenner αποσβένει κάθε σχέδιο στο αναφερόμενο **αρχικό** επιτόκιο, οπότε οι ρυθμίσεις `CAPS`/`MAXRATE`/`WORSTCASE` του ARM και τα βήματα `BDRATES` της μείωσης επιτοκίου εμφανίζονται στη λίστα ως όροι του δανείου, αλλά **δεν** εφαρμόζονται στον υπολογισμό της δόσης — τα παρακάτω στοιχεία του ARM και της μείωσης επιτοκίου αντικατοπτρίζουν τα αρχικά τους επιτόκια 4.75% και 4.50% διατηρούμενα σταθερά για όλη τη διάρκεια. Αντιμετωπίστε αυτά τα δύο σύνολα ως κόστη βέλτιστης περίπτωσης (αρχικού επιτοκίου), όχι ως ανώτατα όρια χειρότερης περίπτωσης.

> **Πρόσθετη σημείωση εκτέλεσης.** Στην τρέχουσα έκδοση του μηχανισμού, το μεμονωμένο μπλοκ σύνοψης "Loan #3" αναφέρει έναν διαφορετικό (υψηλότερο, με βάση τη χειρότερη περίπτωση CAPS/MAXRATE) συνολικό τόκο για το ARM από αυτόν που εμφανίζεται στον πίνακα COMPARE στους 120 μήνες. Η ασυνέπεια αναπαράγεται απαράλλακτη και στο αμετάφραστο αγγλικό notebook με την τρέχουσα έκδοση του jenner (δεν εμφανιζόταν στην αποθηκευμένη έξοδο του αγγλικού notebook, άρα πρόκειται για μεταγενέστερη μετατόπιση της έκδοσης του μηχανισμού, όχι ζήτημα της μετάφρασης). Η ανάλυση εδώ βασίζεται στα στοιχεία του πίνακα COMPARE ($7,099.76), όπως και η αρχική αγγλική αφήγηση.

In [3]:
/* NOTE (engine bug worked around): PROC LOAN's per-loan LABEL= value
   is byte-truncated internally (src/runtime/procs/loan.rs truncate())
   without respecting UTF-8 character boundaries, which panics on
   multi-byte Greek text ("end byte index is not a char boundary").
   The four plan labels below are kept in English/ASCII to avoid the
   crash -- everything else in this notebook is localized to Greek. */
ΔΙΑΔΙΚΑΣΙΑ loan START=2026:7 amount=27500 life=120 outsum=plan_sum;

   fixed rate=6.53
         ΕΤΙΚΕΤΑ='Federal Direct Unsubsidized (Standard)';

   fixed rate=5.99 init=412.50
         ΕΤΙΚΕΤΑ='Private Refinance (with fee)';

   arm rate=4.75 caps=(1 5) maxrate=9.75 adjustfreq=12 worstcase
       ΕΤΙΚΕΤΑ='Private Variable ARM (worst case)';

   buydown rate=4.50 bdrates=(13=5.50 25=6.50)
           ΕΤΙΚΕΤΑ='Employer 2-1 Buydown then 6.5%';

   COMPARE at=(36 60 120) ALL taxrate=22 marr=4 OUTCOMP=plan_cmp;
ΕΚΤΕΛΕΣΗ;

                            The LOAN Procedure

  Number of loans evaluated:    4

  Loan #1: Federal Direct Unsubsidized (Standard)
    Loan Type:                   Fixed
    Amount (Principal):                27500.00
    Interest Rate (annual %):            6.5300
    Life (months):                          120
    Monthly Payment:                     312.68
    Total Paid (P&I):                  37521.22
    Total Interest:                    10021.22

  Loan #2: Private Refinance (with fee)
    Loan Type:                   Fixed
    Amount (Principal):                27500.00
    Interest Rate (annual %):            5.9900
    Life (months):                          120
    Monthly Payment:                     305.17
    Total Paid (P&I):                  36620.20
    Total Interest:                     9120.20
    Initialization Cost:                 412.50

  Loan #3: Private Variable ARM (worst case)
    Loan Type:                   Adjustable Rate
    Amount (Principal):      


NOTE: PROC LOAN loan analysis

NOTE: PROC LOAN statement used.


## Βήμα 4 — Επανεκτέλεση του συνιστώμενου σταθερού σχεδίου μεμονωμένα

Για τον δανειολήπτη που εκτιμά τη βεβαιότητα της δόσης, το ομοσπονδιακό σταθερού επιτοκίου βασικό σχέδιο είναι η ασφαλής προεπιλογή. Το επανεκτελούμε μεμονωμένα για να επιβεβαιώσουμε τα βασικά οικονομικά του στοιχεία: την ίδια μηνιαία δόση **$312.68**, το συνολικό ποσό **$37,521.22** και τον συνολικό τόκο **$10,021.22** που είδαμε στη σύγκριση τεσσάρων σχεδίων, τώρα παρουσιασμένα ως αυτοτελής περίληψη δανείου.

In [4]:
ΔΙΑΔΙΚΑΣΙΑ loan START=2026:7 amount=27500 rate=6.53 life=120 schedule=yearly;
   fixed ΕΤΙΚΕΤΑ='Federal Direct Unsubsidized (Standard)';
ΕΚΤΕΛΕΣΗ;

                            The LOAN Procedure

  Number of loans evaluated:    1

  Loan #1: Federal Direct Unsubsidized (Standard)
    Loan Type:                   Fixed
    Amount (Principal):                27500.00
    Interest Rate (annual %):            6.5300
    Life (months):                          120
    Monthly Payment:                     312.68
    Total Paid (P&I):                  37521.22
    Total Interest:                    10021.22

  Loan Repayment Schedule: Federal Direct Unsubsidized (Standard)
      Year    Begin Balance        Payment       Interest      Principal      End Balance
    ------ ---------------- -------------- -------------- -------------- ----------------
         1         27500.00        3752.12        1736.12        2016.00         25484.00
         2         25484.00        3752.12        1600.47        2151.66         23332.34
         3         23332.34        3752.12        1455.68        2296.44         21035.90
         4         2103


NOTE: PROC LOAN loan analysis

NOTE: PROC LOAN statement used.


## Ερμηνεία των αποτελεσμάτων

Και τα τέσσερα σχέδια αποσβένουν πλήρως το ίδιο κεφάλαιο των $27,500 σε 120 μήνες (το ανεξόφλητο υπόλοιπο κάθε σχεδίου φτάνει τα $0.00 τον μήνα 120 στον πίνακα COMPARE), οπότε η απόφαση εξαρτάται από τη **μηνιαία δόση** και τον **συνολικό τόκο στο αναφερόμενο επιτόκιο**:

| Σχέδιο | Επιτόκιο | Δόση | Συνολικός Τόκος | Σημειώσεις |
|------|---------|------|----------------|-------|
| Ομοσπονδιακό Άμεσο Βασικό | 6.53% | $312.68 | $10,021.22 | Χωρίς προμήθεια· ισχυρότερες προστασίες δανειολήπτη |
| Ιδιωτική Αναχρηματοδότηση | 5.99% | $305.17 | $9,120.20 | Προμήθεια έναρξης $412.50 |
| Ιδιωτικό Κυμαινόμενο ARM | 4.75% (αρχικό) | $288.33 | $7,099.76 | Το επιτόκιο μπορεί να αυξηθεί· το ποσό είναι το κόστος στο αρχικό επιτόκιο |
| Μείωση Επιτοκίου Εργοδότη 2-1 | 4.50% (αρχικό) | $285.01 | $6,700.67 | Εξαρτάται από τη συνέχιση της απασχόλησης |

- Το **ομοσπονδιακό βασικό** σχέδιο είναι το ακριβότερο σε τόκο ($10,021.22) αλλά προσφέρει μια σταθερή, προβλέψιμη δόση $312.68 χωρίς προμήθεια.
- Η **ιδιωτική αναχρηματοδότηση** μειώνει το επιτόκιο στο 5.99% ($9,120.20 τόκος, $901 λιγότερο από το ομοσπονδιακό σχέδιο) αλλά χρεώνει προμήθεια έναρξης $412.50, οπότε το καθαρό της πλεονέκτημα έναντι του ομοσπονδιακού σχεδίου είναι περίπου $488 τόκου μείον προμήθεια — σημαντικό μόνο αν το δάνειο διαρκέσει αρκετά ώστε να μην αναχρηματοδοτηθεί.
- Το **ARM** και η **μείωση επιτοκίου** εμφανίζουν εδώ τον χαμηλότερο τόκο ($7,099.76 και $6,700.67) απλώς επειδή τα **αρχικά** τους επιτόκια είναι πολύ χαμηλότερα από τις σταθερές προσφορές. Όπως σημειώθηκε στο Βήμα 3, το Jenner διατηρεί αυτά τα αρχικά επιτόκια σταθερά, οπότε αυτά είναι στοιχεία βέλτιστης περίπτωσης: ένα πραγματικό ARM που αναπροσαρμόζεται προς τα πάνω — ή μια μείωση επιτοκίου που χάνει την επιδότηση του εργοδότη — θα κατέληγε υψηλότερα, και η δόση δεν είναι εγγυημένη.

Ο πίνακας `COMPARE` ενισχύει αυτό δείχνοντας πόσο γρήγορα αποσβένεται κάθε σχέδιο. Στους 36 μήνες, το ομοσπονδιακό σχέδιο έχει καταβάλει $4,792.27 τόκου και έχει εξοφλήσει $6,464.10 κεφαλαίου (υπόλοιπο $21,035.90), ενώ η μείωση επιτοκίου έχει καταβάλει μόνο $3,263.97 τόκου και έχει εξοφλήσει $6,996.24 κεφαλαίου (υπόλοιπο $20,503.76) — τα σχέδια χαμηλότερου επιτοκίου κοστίζουν λιγότερο σε τόκο και μειώνουν το κεφάλαιο ταχύτερα τα πρώτα τρία χρόνια. Για έναν αποφοιτούντα με αποστροφή στον κίνδυνο, η βεβαιότητα επιτοκίου του ομοσπονδιακού βασικού σχεδίου συχνά δικαιολογεί τον υψηλότερο τόκο του· για έναν δανειολήπτη σίγουρο για σταθερή, διαρκή απασχόληση, η επιδοτούμενη έναρξη της μείωσης επιτοκίου αποφέρει τις μεγαλύτερες εξοικονομήσεις ανάμεσα στις επιλογές που πραγματικά σταθεροποιούν το επιτόκιό τους.